### PR TITLE
Allow empty array

### PR DESCRIPTION
**Pull request by islomar**
Currently, the `ConvertTFstate` is deleting keys of type arrays which are empty, without checking the `allowEmptyValues` argument.

This commit is intended to fix that.

https://github.com/GoogleCloudPlatform/terraformer/issues/137

### CODE REVIEW

**Comment by islomar** [comment]
I signed it!

> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fterraformer%2Fpull%2F140) for more info**.

I signed it!

File: terraform_utils/resource.go [hunk 1]
```diff
@@ -86,7 +86,16 @@ func (r *Resource) ConvertTFstate() {
 	// delete empty array
 	for key := range r.InstanceState.Attributes {
 		if strings.HasSuffix(key, ".#") && r.InstanceState.Attributes[key] == "0" {
-			delete(attributes, key)
+			allowEmptyValue := false
+			for _, allowedEmptyValue := range r.AllowEmptyValues {
+				if allowedEmptyValue == key {
```
Comment (sergeylanzman) [inline terraform_utils/resource.go]: `allowedEmptyValue` it's regex pattern.
Maybe better move `AllowEmptyValues` check to separate function and call to function from 2 places

**Comment by sergeylanzman** [comment]
please remove not related commits

**Comment by islomar** [comment]
> please remove not related commits

Done

File: terraform_utils/resource.go (hunk 1 above)
Comment (islomar) [reply terraform_utils/resource.go]: Well seen, done!

**Pull request merged**